mentat/python_client/client.py

```diff
@@ -29,25 +29,36 @@ def __init__(
 
         self._tasks: Set[asyncio.Task[None]] = set()
         self.acc_task: asyncio.Task[None] | None = None
-        self._started = False
+        self.started = False
         self._accumulated_message = ""
 
     async def call_mentat(self, message: str):
-        if not self._started:
-            await self._startup()
+        if not self.started:
+            await self.startup()
         assert isinstance(self.session, Session), "Client is not running"
         await self.session.stream.send(
             message,
             source=StreamMessageSource.CLIENT,
             channel=f"input_request:{self._input_request_message.id}",
         )
-        self._input_request_message = await self.session.stream.recv("input_request")
+        # If the Session shuts down, we'll be waiting here forever;
+        # but because we need to wait for this request to insure that all changes have been made,
+        # setting a timeout is the best way to determine if the Session has shut down
+        try:
+            self._input_request_message = await asyncio.wait_for(
+                self.session.stream.recv("input_request"), 3
```

Comment:
Is 3 enough for you? I get `TImeoutError` 100% of the time but it works if I change it to 300. In theory we should only wait here when the session encounters an exception and shuts down right? It seems fine to bias towards a high number.

Reply:
Actually, I think I might just add in the stream message that indicates that the changes are finished now

Reply:
I'm a little confused. Shouldn't the input_request signal the edits have been made? And isn't a number greater than the max LLM response time be necessary here? Which should be at least 30? Session exits should be rare so I'd bias towards not killing sessions that would have succeeded

Reply:
I completely changed this, now I have the stream send a message in a different channel signaling that the edits are completed and we listen to that instead of waiting for an input request message (which always felt a bit hacky anyways).